testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -7189,6 +7189,33 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
         "INTEGER ARRAY NOT NULL");
   }
 
+  @Test void testMapContainsKeyFunc() {
```

Comment:
Make sure result as same as spark-sql. 
And can you add more negative tests?

Reply:
Add docs for this test

Reply:
Negative tests refer to error tests. I know that there is a certain problem with type conversion here. map_contains_key(map(1, 2), cast(1 as double) returns false, and spark returns true. This may involve some node type conversion bug

Reply:
In this case this looks like a spark bug

Reply:
So I'm not sure if I want to fix it, I will open a jira to discuss it

Reply:
https://issues.apache.org/jira/browse/CALCITE-6235